ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -504,6 +504,18 @@ def test_source_ipahealthcheck_meta_core_metacheck(self):
         assert data[0]["kw"]["ipa_version"] in result.stdout_text
         assert data[0]["kw"]["ipa_api_version"] in result.stdout_text
 
+    def test_fips_mode_check_exists(self):
+        """
+        Testcase to verify whether FIPS mode check exists
+        in ipahealthcheck.meta.core
+        """
+        returncode, data = run_healthcheck(self.master,
+                                           "ipahealthcheck.meta.core")
+        assert returncode == 0
+        for check in data:
+            if check["kw"].get("fips", None):
```

Comment:
If the check doesn't return any value for "fips", the test will be successful. Is it really what you want? I would rather check explicitly that there is a value, and also check that the returned value is consistent with the system.
There are multiple ways to check if the system is fips-enabled:
- the command fips-mode-setup --is-enabled
- the content of the file /proc/sys/crypto/fips_enabled
- the output of `sysctl crypto.fips_enabled`

Reply:
I suspect it would be worthwhile to also execute this test with fips_mode = True

Reply:
I suspect you are right, Rob. I'll extend the test.

Reply:
@miskopo if you want to run in fips mode, you can have a look at the test in `test_integration/test_fips.py` that is faking fips mode.

Reply:
To be perfectly candid, I am not sure how to proceed. From my understanding, the base class in `test_fips` only fakes FIPS mode by returning `true` to `fips_enabled` queries.

Reply:
Right. It might be handy for pr-ci to have a fips option in the YAML but for now I think, since this is a small and very-confined test, to duplicate it in test_fips. The FIPS version can be simplified since you can assert that healthcheck returns fips=True. You may need to do additional setup to ensure the healthcheck package is installed. This is normally handled in the healthcheck test class. I'd add a cross-reference to each one so we know that there is some duplication, and why.

Reply:
Okay, thank you, Rob. I'll try to duplicate relevant bits and see what can be done. Also, with the FIPS mode - I think I may be able to implement FIPS param support to PR-CI.